activerecord/lib/active_record/testing/query_assertions.rb

```diff
@@ -72,9 +72,9 @@ def assert_queries_match(match, count: nil, include_schema: false, &block)
           matched_queries = queries.select { |query| match === query }
 
           if count
-            assert_equal count, matched_queries.size, "#{matched_queries.size} instead of #{count} queries were executed.#{queries.empty? ? '' : "\nQueries:\n#{queries.join("\n")}"}"
+            assert_equal count, matched_queries.size, "#{matched_queries.size} instead of #{count} matching queries were executed#{queries.empty? ? '' : ". Queries:\n\n#{queries.join("\n\n")}"}"
           else
-            assert_operator matched_queries.size, :>=, 1, "1 or more queries expected, but none were executed.#{queries.empty? ? '' : "\nQueries:\n#{queries.join("\n")}"}"
+            assert_operator matched_queries.size, :>=, 1, "1 or more matching queries expected, but none were executed#{queries.empty? ? '' : ". Queries:\n\n#{queries.join("\n\n")}"}"
```

Comment:
Here however there might have been non-matching queries, so we need to keep the print logic. I also specified 'matching queries' in the message to remove any confusion.